react-intro/react-training-es6-intro.md

```diff
@@ -0,0 +1,174 @@
+
+___
+
+## [fit] Our old friend __*var*__
+
+We've all used __*var*__ but it causes a lot of issues. It can be *redifined* and __updated__, which causes a lot of __*bugs*__. 
+
+The scope of a variable declared with var is its __*current execution context*__, which is either the enclosing function or, for variables declared outside any function, global. 
+
+Examples: https://tinyurl.com/ycsn2ssa
+___
+
+## [fit] __*let*__ and __*const*__
+
+These are *block* __scoped__ variables. (__*i.e.*__ a "__*block*__" is in curly brackets)
+
+Variables declared by let have their __scope__ in the __*block*__ for which they are defined, as well as in any contained __*sub-blocks*__. In this way, let works very much like __*var*__. The main difference is that the __scope__ of a __*var*__ variable is the entire enclosing function.
+
+___
+
+## [fit] __*let*__ examples
+
+Using __*let*__ to define the same variable twice in the same block, will give you an error: "Identifier 'x' has already been declared".
+
+You can, however, redeclare it in a sub block.
+
+Examples: https://codepen.io/slafleche/pen/odPBrV
+
+___
+
+## [fit] __*const*__ examples
+
+__*const*__ have the same block scoping as __*let*__. It is not immutable, but the value of a constant cannot change through re-assignment, and it can't be redeclared.
+
+Examples:
+
+https://codepen.io/slafleche/pen/MGPpmr
+
+___
+
+## [fit] tldr
+
+If you're unsure what to use, start with __*const*__ and see if you get an error. If you do, evaluate if it's your fault or if it makes sense to use __*let*__. 
+
+Don't use __*var*__. *ES6* forces you to have better structured code and you should take advantage of that. 
+
+___
+
+## [fit] Arrow functions *=>*
+
+Replaces __*anonymous functions*__ 
+
+`function () {}` __*replaced with*__ `() => {}`
+
+Examples:
+
+https://codepen.io/slafleche/pen/YLJZgy
+
+___
+
+## [fit] __*Lexical scoping*__ for arrow functions
+
+Lexical Scoping just means that it uses __*this*__ from the code that contains the arrow function.
+
+The thing to pay attention to is the scope of __*this*__ and the function in `setInterval`
+
+https://codepen.io/slafleche/pen/aGRWmo
+
+___
+
+## [fit] __*forEach*__ and __*map*__
+
+__*forEach*__ : regular loop through items.
+
+__*map*__ : returns new array, applying a transformation on each element. You can also chain the results to other array 
+functions like __*filter()*__ or __*reverse()*__.
+
+Examples: https://codepen.io/slafleche/pen/LmgdmQ
+
+___
+
+## [fit] __*spread*__ operators
+
+Passes each value inside array or object with a handy syntax. Important to note that the spread operator *does not* do a deep copy. While the spread operator __does__ create a new object, the properties’ values are simply references and not new instances
+
+Examples: https://codepen.io/slafleche/pen/pVxLmE
+
+___
+
+## [fit] Template literals
+
+Built in templating style strings! Just use back ticks "__*\`*__".
+
+Insert varaibles with *`${vaName}`*.
```

Comment:
varaibles -> variables.

Also, it might be more accurate to say expressions here. In PHP template strings you can only insert variables, but in the JS template strings you can use expressions such as a function call, array access, etc.